### ISSUE TITLE
Handling of unsupported types

### ISSUE THREAD

**Issue by dvicini** [csingley/ibflex#54]
Hi,

I've just started to play around with this package. One thing I encountered (as many others it seems) is that the parser fails violently if the flex query results contains unknown fields. I was wondering if there would be a way to change this behavior, maybe optionally, to simply ignore unknown fields? The IBKR web interface allows to very easily add all available fields to a query, and manually figuring out which to enable for the parser and which to disable appears a bit tricky. Many of the unsupported fields seem to be details that are maybe not interesting in most use cases, so it probably would make sense for the code to fail only once the user tries to access such an attribute.

Any thoughts on this? I guess this is more of a general feature request than a specific issue.

**Comment by csingley** [comment]
I mean, you could just [check in the new types](https://github.com/csingley/ibflex/blob/master/ibflex/Types.py).  I've gone to some trouble to make adding types as easy as possible.... surely easier than logging into IBKR web app and twiddling the query?

Parser strictness is a feature not a bug in my view.  It just doesn't come free, when IBKR SWEs are constantly at work extending the schema willy-nilly.

**Comment by dvicini** [comment]
Yes I agree, adding new types is indeed straightforward and easier than twiddling the query in the slow web interface. I also agree that it's a bit frustrating that IBKR constantly extends their schema it seems. Hence my suggestion of simply ignoring unknown fields. Again, you are right that a parser generally should be strict to prevent bugs, but I think this can be relaxed a bit when it comes to a query having fields that are new and unknown. I don't think that ignoring new fields is much less robust than what happens now, i.e. new types having to be added to the classes gradually (without much testing etc.). It's not like there is much logic in these type declarations to begin with. One could potentially even automatically generate these type declarations from a downloaded example query result.

In the end up to you, this is your project :)

**Comment by csingley** [comment]
Autogenerating the types would absolutely be the move - unfortunately there's just not enough there for a machine to reasonably inter the type of data primitives (`str` vs `Decimal`, or god forbid `Enum`).

**Comment by csingley** [comment]
The right way to do this is to get devs from a few funds & RIAs, who in aggregate can speak for O(US$10^8) net assets, on the same page to drive this change with the broker.

**Comment by agusalex** [comment]
I drafted [PR](https://github.com/csingley/ibflex/pull/59) to address some of this drift, at least for Paxos integration right now. Thanks for the good work!

**Comment by csingley** [comment]
And thank you for putting in the work here - sorry for the delay in reviewing.

**Comment by csingley** [comment]
Fixed in 8c9c49f — the parser is now lenient and warns on unknown attributes/elements instead of raising.
